CAP/gap/Derivations.gi

```diff
@@ -348,28 +358,32 @@ InstallMethod( AddDerivationPair,
   function( graph, target_op1, target_op2, used_ops_with_multiples,
             implementations_with_extra_filters1, implementations_with_extra_filters2 )
     local weight, category_filter, description, derivation1, derivation2, collected_list,
-          operations_in_graph, current_list, current_implementation, loop_multiplier;
+          operations_in_graph, current_list, current_implementation, loop_multiplier,
+          preconditions_complete;
     
     weight := CAP_INTERNAL_RETURN_OPTION_OR_DEFAULT( "Weight", 1 );
     category_filter := CAP_INTERNAL_RETURN_OPTION_OR_DEFAULT( "CategoryFilter", IsCapCategory );
     description := CAP_INTERNAL_RETURN_OPTION_OR_DEFAULT( "Description", "" );
     loop_multiplier := CAP_INTERNAL_RETURN_OPTION_OR_DEFAULT( "WeightLoopMultiple", 2 );
+    preconditions_complete := CAP_INTERNAL_RETURN_OPTION_OR_DEFAULT( "ConditionsListComplete", false );
     
     ## get used ops
     ## Is this the right place? Or should this only be done when no ops are given?
     operations_in_graph := Operations( graph );
     
     collected_list := [ ];
     
-    for current_implementation in Concatenation( implementations_with_extra_filters1, implementations_with_extra_filters2 ) do
-        
-        current_list := CAP_INTERNAL_FIND_APPEARANCE_OF_SYMBOL_IN_FUNCTION( current_implementation[ 1 ], operations_in_graph, loop_multiplier );
-        current_list := List( current_list, i -> [ ValueGlobal( i[ 1 ] ), i[ 2 ] ] );
-        ## TODO: Maybe move the next line to AddDerivationPairToCAP
-        current_list := Concatenation( current_list, CAP_INTERNAL_FIND_APPEARANCE_OF_SYMBOL_IN_FUNCTION_FOR_MONOIDAL_CATEGORIES( current_implementation[ 1 ], loop_multiplier ) );
-        collected_list := CAP_INTERNAL_MERGE_PRECONDITIONS_LIST( collected_list, current_list );
-        
-    od;
+    if preconditions_complete = false then
```

Comment:
Your documentation above says that every other value
than `true` will be interpreted as non-complete, not only
`false.`

Reply:
Thank you, yes. I changed the logic again afterwards. I will fix up the documentation.